espnet2/train/trainer.py

```diff
@@ -709,7 +731,7 @@ def train_one_epoch(
                 for iopt, optimizer in enumerate(optimizers):
                     if optim_idx is not None and iopt != optim_idx:
                         continue
-                    optimizer.zero_grad()
+                    optimizer.zero_grad(set_to_none=True)
```

Comment:
This is the recommended setting by PyTorch now, it is both faster and more memory efficient. However, it may also slightly affect the learning curve